app-backend/db/src/main/scala/com/azavea/rf/database/TemplateVersionDao.scala

```diff
@@ -0,0 +1,82 @@
+package com.azavea.rf.database
+
+import java.sql.Timestamp
+
+import com.azavea.rf.database.Implicits._
+import com.azavea.rf.datamodel._
+import doobie._
+import doobie.implicits._
+import doobie.postgres._
+import doobie.postgres.implicits._
+import cats._
+import cats.data._
+import cats.effect.IO
+import cats.implicits._
+import java.util.UUID
+
+import scala.concurrent.Future
+
+object TemplateVersionDao extends Dao[TemplateVersion] {
+  val tableName = "template_versions"
+
+  val selectF = sql"""
+    SELECT
+    FROM """ ++ tableF
+
+  def insert(
+    tvCreate: TemplateVersion.Create
+  ): ConnectionIO[TemplateVersion] = {
+    sql"""
+      INSERT INTO ${tableName}
+      (created_at, modified_at, created_by, modified_by, version, description, changelog, template_id, analysis_id)
+      VALUES
+      (${tvCreate.createdAt}, ${tvCreate.modifiedAt}, ${tvCreate.createdBy}, ${tvCreate.modifiedBy},
+      ${tvCreate.version}, ${tvCreate.description}, ${tvCreate.changelog}, ${tvCreate.templateId},
+      ${tvCreate.analysisId})
+    """.update.withUniqueGeneratedKeys[TemplateVersion](
+      "id", "created_at", "modified_at", "created_by", "modified_by", "version", "description", "changelog",
+      "template_id", "analysis_id"
+    )
+  }
+
+  def getById(template: Template, templateVersionId: Long, user: User):
+      ConnectionIO[Option[TemplateVersion.WithRelated]] = {
+    query
+      .filter(fr"template_id = ${template.id}")
```

Comment:
id filter should work here

Reply:
nope, `.filter(UUID)` creates a fragment in form `fr"id = ${id}`, and this one's `template_id`